api/v1alpha1/network_types.go

```diff
@@ -50,6 +50,14 @@ type NetworkSpec struct {
 type NetworkStatus struct {
 }
 
+// NetworkReference is an object that points to a Network.
+type NetworkReference struct {
+	// Name of the Network resource being referenced.
+	Name string `json:"name"`
+	// APIVersion of the referent.
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
Did you intend to have this `omitempty`?

Reply:
Yes, this pattern was copied from `NetworkInterfaceProviderReference`. Is that pattern incorrect? I can add the corresponding kubebuilder optional comment.

Reply:
The way I was reading it... if we wanted to point to a different type we'd create a new type reference matching that type. Not sure how flawed that thinking is.

Reply:
A `Reference` is flexible so it can point to whatever type. I guess what you're saying this will always reference the NetOP `Network` type?

If you want to keep this `omitempty` then add the `// +optional` convention.

Reply:
Right. It's built into the name. IOW: it's not clear that it's valid to ever reference another type of `Network`. If we decided to make that available, we could just rev the API and add the backward-compatible kind parameter at that point.

Happy to add the kubebuilder tag.